common/work.c

```diff
@@ -9,7 +9,7 @@
 #include <work.h>
 
 /* Work item queue capacity. */
-#define MAX_WORK_ITEMS 1
+#define MAX_WORK_ITEMS 2
```

Comment:
Why are you choosing 2? This should be able to handle more than just two if the work items are being added faster than they are being processed